### ISSUE TITLE
Git Credentials Not Found: Desktop Application

### ISSUE THREAD

**Issue by mohilkhare1708** [desktop/desktop#2747]
![screenshot 56](https://user-images.githubusercontent.com/22223027/30498033-d7824fd8-9a72-11e7-9842-c1c19631b622.png)
Greetings! I have been using Github to share my code with my teachers using the **Desktop Application** and it worked fine for me. I use **VS Code** as my code editor and today while I was pushing my code, I received the following message.

> Git credentials for https://github.com/mohilkhare1708/Friend-Finder.git not found:
exit status 1
error: failed to push some refs to 'https://github.com/mohilkhare1708/Friend-Finder.git'

I'm not able to understand what this message means and what am I supposed to do here. I have always been able to do the same thing without any issue and today all of a sudden this happened.

I haven't even touched the .git folder as well.

Can someone please be kind enough to explain me how to fix it in an easy language?

Thanks a lot!

Mohil

**Comment by joshaber** [comment]
Hi @mohilkhare1708! Huh, that's an interesting error. A few questions to help us figure out what's going on here:

1. What version of GitHub Desktop are you running? This is displayed under the `About GitHub Desktop` menu item.
2. Could you check `%APPDATA%\GitHub Desktop\logs\` to see if there are any logs and drop them in here if there are?
3. Are you using Git LFS?

**Comment by AndyBPA** [comment]
I'm also getting the same issue after upgrading to v0.91 (worked fine in previous version) - LOG error:

2017-09-16T00:48:56.053Z - error: [ui] `git -c credential.helper= push origin release-0.98:release-0.98 --progress` exited with an unexpected code: 1.
Remote "origin" does not support the LFS locking API. Consider disabling it with:
  $ git config lfs.https://github.com/user/project.git/info/lfs.locksverify false

Git credentials for https://github.com/user/project.git not found:
exit status 1
error: failed to push some refs to 'https://github.com/user/project.git'

We are using LFS.

**Comment by mohilkhare1708** [comment]
joshaber thanks for responding! Here are the things you've asked for:

> What version of GitHub Desktop are you running? This is displayed under the About GitHub Desktop menu item.

Mine is 0.9.1

> Could you check %APPDATA%\GitHub Desktop\logs\ to see if there are any logs and drop them in here if there are?


```
2017-09-14T07:29:33.861Z - warn: [ui] Unknown URL action: unknown - payload: {"name":"unknown","url":"--squirrel-firstrun"}
2017-09-14T07:29:35.626Z - info: [ui] Executing installGlobalLFSFilter: git lfs install --skip-repo (took 1.733s)
2017-09-14T07:30:27.140Z - info: [ui] Opt out reported.
2017-09-14T07:32:17.458Z - info: [ui] Executing push: git -c credential.helper= push origin master --set-upstream --progress (took 15.070s)
2017-09-14T07:32:21.289Z - info: [ui] Executing fetch: git -c credential.helper= fetch --progress --prune origin (took 3.830s)
2017-09-14T07:32:27.176Z - info: [ui] Executing fetch: git -c credential.helper= fetch --progress --prune origin (took 3.758s)
2017-09-14T07:34:29.186Z - info: [ui] Executing fetch: git -c credential.helper= fetch --progress --prune origin (took 3.798s)
2017-09-14T07:34:56.142Z - info: [ui] Executing fetch: git -c credential.helper= fetch --progress --prune origin (took 3.853s)
2017-09-14T07:35:14.064Z - info: [ui] Executing fetch: git -c credential.helper= fetch --progress --prune origin (took 3.827s)
2017-09-14T07:38:54.393Z - info: [ui] Executing push: git -c credential.helper= push origin master:master --progress (took 15.757s)
2017-09-14T07:39:00.065Z - info: [ui] Executing fetch: git -c credential.helper= fetch --progress --prune origin (took 5.669s)
2017-09-14T07:40:47.162Z - info: [ui] Executing fetch: git -c credential.helper= fetch --progress --prune origin (took 4.109s)
2017-09-14T07:46:21.380Z - info: [ui] Executing fetch: git -c credential.helper= fetch --progress --prune origin (took 4.316s)
2017-09-14T07:51:55.141Z - info: [ui] Executing fetch: git -c credential.helper= fetch --progress --prune origin (took 4.086s)
2017-09-14T07:57:28.943Z - info: [ui] Executing fetch: git -c credential.helper= fetch --progress --prune origin (took 3.872s)
2017-09-14T08:02:53.702Z - info: [ui] Executing push: git -c credential.helper= push origin master:master --progress (took 13.553s)
2017-09-14T08:02:57.583Z - info: [ui] Executing fetch: git -c credential.helper= fetch --progress --prune origin (took 3.883s)
2017-09-14T08:03:01.749Z - info: [ui] Executing fetch: git -c credential.helper= fetch --progress --prune origin (took 3.836s)
2017-09-14T13:54:03.339Z - info: [ui] Executing getBranches: git for-each-ref --format=%(refname)%00%(refname:short)%00%(upstream:short)%00%(objectname)%00%(author)%00%(parent)%00%(subject)%00%(body)%00%1F refs/heads refs/remotes (took 1.891s)
2017-09-14T13:54:03.355Z - info: [ui] Executing getStatus: git status --untracked-files=all --branch --porcelain=2 -z (took 1.989s)
2017-09-14T13:54:03.370Z - info: [ui] Executing getRecentBranches: git log -g --no-abbrev-commit --pretty=oneline HEAD -n 2500 -- (took 1.943s)
2017-09-14T13:54:05.062Z - info: [ui] Executing installGlobalLFSFilter: git lfs install --skip-repo (took 1.881s)
2017-09-14T13:54:12.328Z - info: [ui] Executing fetch: git -c credential.helper= fetch --progress --prune origin (took 4.402s)
2017-09-14T13:59:22.229Z - info: [ui] Executing fetch: git -c credential.helper= fetch --progress --prune origin (took 4.022s)
2017-09-14T14:04:32.247Z - info: [ui] Executing fetch: git -c credential.helper= fetch --progress --prune origin (took 4.016s)
2017-09-14T14:09:42.694Z - info: [ui] Executing fetch: git -c credential.helper= fetch --progress --prune origin (took 4.486s)
2017-09-14T14:14:53.549Z - info: [ui] Executing fetch: git -c credential.helper= fetch --progress --prune origin (took 4.348s)
2017-09-14T14:20:06.339Z - info: [ui] Executing fetch: git -c credential.helper= fetch --progress --prune origin (took 6.134s)
2017-09-14T14:25:16.174Z - info: [ui] Executing fetch: git -c credential.helper= fetch --progress --prune origin (took 3.969s)
2017-09-14T14:30:30.584Z - info: [ui] Executing fetch: git -c credential.helper= fetch --progress --prune origin (took 6.365s)
2017-09-14T14:35:41.397Z - info: [ui] Executing fetch: git -c credential.helper= fetch --progress --prune origin (took 4.186s)
2017-09-14T14:40:51.109Z - info: [ui] Executing fetch: git -c credential.helper= fetch --progress --prune origin (took 3.900s)
2017-09-14T14:46:02.870Z - info: [ui] Executing fetch: git -c credential.helper= fetch --progress --prune origin (took 5.649s)
2017-09-14T14:51:15.358Z - info: [ui] Executing fetch: git -c credential.helper= fetch --progress --prune origin (took 4.169s)
2017-09-14T14:56:23.416Z - error: [ui] `git -c credential.helper= fetch --progress --prune origin` exited with an unexpected code: 128.
fatal: unable to access 'https://github.com/mohilkhare1708/Friend-Finder.git/': Couldn't resolve host 'github.com'

2017-09-14T14:56:23.758Z - warn: [ui] getFetchPollInterval: failed for mohilkhare1708/Friend-Finder
TypeError: Failed to fetch
2017-09-14T15:56:32.108Z - info: [ui] Executing fetch: git -c credential.helper= fetch --progress --prune origin (took 3.910s)
2017-09-14T16:01:42.403Z - info: [ui] Executing fetch: git -c credential.helper= fetch --progress --prune origin (took 4.201s)
2017-09-14T16:06:52.364Z - info: [ui] Executing fetch: git -c credential.helper= fetch --progress --prune origin (took 4.155s)
2017-09-14T16:12:02.598Z - info: [ui] Executing fetch: git -c credential.helper= fetch --progress --prune origin (took 4.391s)
2017-09-14T16:17:13.031Z - info: [ui] Executing fetch: git -c credential.helper= fetch --progress --prune origin (took 3.821s)
2017-09-14T16:22:23.084Z - info: [ui] Executing fetch: git -c credential.helper= fetch --progress --prune origin (took 3.873s)
2017-09-14T16:27:33.028Z - info: [ui] Executing fetch: git -c credential.helper= fetch --progress --prune origin (took 3.820s)
2017-09-14T16:33:42.180Z - info: [ui] Executing fetch: git -c credential.helper= fetch --progress --prune origin (took 62.972s)
2017-09-14T16:33:42.196Z - error: [ui] `git -c credential.helper= fetch --progress --prune origin` exited with an unexpected code: 128.
fatal: unable to access 'https://github.com/mohilkhare1708/Friend-Finder.git/': schannel: failed to receive handshake, SSL/TLS connection failed

2017-09-14T16:33:42.528Z - warn: [ui] getFetchPollInterval: failed for mohilkhare1708/Friend-Finder
TypeError: Failed to fetch
2017-09-14T17:33:51.822Z - info: [ui] Executing fetch: git -c credential.helper= fetch --progress --prune origin (took 4.577s)
2017-09-14T17:39:02.393Z - info: [ui] Executing fetch: git -c credential.helper= fetch --progress --prune origin (took 4.173s)
2017-09-14T17:44:13.283Z - info: [ui] Executing fetch: git -c credential.helper= fetch --progress --prune origin (took 4.077s)
2017-09-14T17:49:23.685Z - info: [ui] Executing fetch: git -c credential.helper= fetch --progress --prune origin (took 4.480s)
2017-09-14T17:54:34.423Z - info: [ui] Executing fetch: git -c credential.helper= fetch --progress --prune origin (took 4.218s)
2017-09-14T17:59:44.012Z - info: [ui] Executing fetch: git -c credential.helper= fetch --progress --prune origin (took 3.812s)
2017-09-14T18:04:54.095Z - info: [ui] Executing fetch: git -c credential.helper= fetch --progress --prune origin (took 3.894s)
2017-09-14T18:10:05.963Z - info: [ui] Executing fetch: git -c credential.helper= fetch --progress --prune origin (took 5.757s)
2017-09-14T18:15:17.198Z - info: [ui] Executing fetch: git -c credential.helper= fetch --progress --prune origin (took 4.994s)
2017-09-14T18:20:27.665Z - info: [ui] Executing fetch: git -c credential.helper= fetch --progress --prune origin (took 4.459s)
2017-09-14T18:23:40.821Z - info: [ui] Executing createCommit: git commit -F - (took 1.571s)
2017-09-14T18:23:48.014Z - error: [ui] `git -c credential.helper= push origin master:master --progress` exited with an unexpected code: 128.
fatal: unable to access 'https://github.com/mohilkhare1708/Friend-Finder.git/': Couldn't resolve host 'github.com'

2017-09-14T18:24:32.024Z - info: [ui] Executing push: git -c credential.helper= push origin master:master --progress (took 18.465s)
2017-09-14T18:24:36.441Z - info: [ui] Executing fetch: git -c credential.helper= fetch --progress --prune origin (took 4.415s)
2017-09-14T18:25:38.320Z - info: [ui] Executing fetch: git -c credential.helper= fetch --progress --prune origin (took 4.131s)
```
I have no idea what this means.

> Are you using Git LFS?

Nope.

I hope this helps.
Thanks!

**Comment by t7m** [comment]
Also experiencing this issue since the latest upgrade.

**Comment by joshaber** [comment]
@mohilkhare1708 It looks like there's a variety of network errors happening there. Has your network connection been unreliable?

@t7m Could you provide some more details and your logs?

**Comment by t7m** [comment]
I'm not sure what details you need. I click "push" and I get the same error. I can open a command prompt and push without issue.

Logs:
2017-09-18T13:25:14.604Z - info: [ui] Executing push: git -c credential.helper= push origin master:master --progress (took 1.256s)
2017-09-18T13:25:14.606Z - error: [ui] `git -c credential.helper= push origin master:master --progress` exited with an unexpected code: 1.
Remote "origin" does not support the LFS locking API. Consider disabling it with:
  $ git config lfs.https://github.com/propertyroomdev/scout-alley.git/info/lfs.locksverify false

Git credentials for https://github.com/propertyroomdev/scout-alley.git not found:
exit status 1
error: failed to push some refs to 'https://github.com/propertyroomdev/scout-alley.git'

**Comment by joshaber** [comment]
@t7m Thanks, that's what I needed! Are you using Git LFS?

**Comment by t7m** [comment]
I have no idea what that is. I installed GitHub desktop and I use the defaults and that's all I know.

**Comment by joshaber** [comment]
That's fine, that's all you _should_ need to know 😄 It sounds like you were effect by a bug from a release or two back.

First, make sure you're running version 0.9.1 and update if you're not.

Then run `rm .git/hooks/pre-push`.

You should be good to go after that.

**Comment by Joel8bitar** [comment]
Hey @joshaber  Really now to Github here! How do i run rm .git/hooks/pre-push from the repository?

Thanks!

**Comment by Joel8bitar** [comment]
@joshaber That helped! Thank you so much!

**Comment by Martincccc** [comment]
@Joel8bitar, Thanks for solving this issue !

**Comment by t7m** [comment]
I wouldn't call that a "fix". What caused the bug? Will the actual issue be fixed?

**Comment by joshaber** [comment]
The actual bug was https://github.com/desktop/desktop/issues/2732 which was fixed in https://github.com/desktop/desktop/pull/2734 and release in 0.9.1.

Users who added a new repository with the buggy release will need to do the [steps above](https://github.com/desktop/desktop/issues/2747#issuecomment-330322002) to fix their affected repositories.

**Comment by t7m** [comment]
"Added a new repository" is not entirely correct. "Cloned an existing repository again" was my issue. I did not add one. (I realize this is essentially the same thing.)

**Comment by joshaber** [comment]
Closing as I believe this was LFS-related and the [steps above](https://github.com/desktop/desktop/issues/2747#issuecomment-330322002) should resolve it.

Please open a new issue if that doesn't fix it!

**Comment by t7m** [comment]
This issue is NOT fixed. I'm on 1.0.0 and the issue CAME BACK.  If I need to run this...

del /f .git\hooks\pre-push

... on every upgrade, the fix is not in place

**Comment by joshaber** [comment]
@t7m I'm sorry to hear that!

I can't seem to reproduce it on 1.0. The steps I'm trying:

1. Add a repository that wasn't in the app already.
1. Try to push/pull. Succeeds.
1. `.git/hooks/pre-push` doesn't exist.

Note that you'll need to do the `del /f .git\hooks\pre-push` for each repository that was added when using the buggy version of the app.

**Comment by AndyBPA** [comment]
After updating to version 1.0 and running 'del /f .git\hooks\pre-push' it seems to be fixed for me.

**Comment by AndyBPA** [comment]
@joshaber Why close the issue? It's still not fixed. I've run that given command on both branches of our repository which fixes it initially. But as soon as you switch branches the issue returns. 

In your testing you should setup two branches then try switching between them to see if you get the issue too (but I think if you've never had the issue, you probably won't get it, which doesn't mean its not there).

The problem is, it only seems to affect a few people, as my associate is not having the same issue as me, he can push fine, and switch branches and push. So it tends to suggest something on my system got out of whack, or doesn't like the latest versions from 0.91 on (as with the other guys here having issues).

I've tried completely uninstalling and reinstalling the latest Client fresh, same problem. Is there saved files / cached files or preferences / settings stored somewhere by the Git Client that I can try clearing out?

**Comment by shiftkey** [comment]
> I've run that given command on both branches of our repository which fixes it initially. But as soon as you switch branches the issue returns.

@AndyBPA could you provide a fresh log file (so we can see the latest errors) as well as the output of this command from `cmd`:

```
dir .git\hooks
```

You mention the problem relating to switching branches, so I want to confirm if there are any other hooks around that are executing that shouldn't be.

**Comment by t7m** [comment]
This issue clearly has not gone away. I'm on 1.0, and I just re-cloned a project. A push still gives me these errors. I have no idea why this bug is closed or why anyone thinks something is fixed. A bug was obviously introduced, and it continues to exist since then. Very disappointing.

**Comment by t7m** [comment]
2017-09-21T02:19:07.639Z - info: [ui] Stats reported.
2017-09-21T02:19:21.385Z - info: [ui] Executing clone: git -c credential.helper= lfs clone --recursive --progress --progress -- https://github.com/[project].git C:\Users\me\Documents\git\[project] (took 13.293s)
2017-09-21T02:24:37.758Z - info: [ui] Executing push: git -c credential.helper= push origin master:master --progress (took 1.913s)
2017-09-21T02:24:37.765Z - error: [ui] `git -c credential.helper= push origin master:master --progress` exited with an unexpected code: 1.
Remote "origin" does not support the LFS locking API. Consider disabling it with:
  $ git config lfs.https://github.com/[project].git/info/lfs.locksverify false

Git credentials for https://github.com/[project].git not found:
exit status 1
error: failed to push some refs to 'https://github.com/[project].git'

**Comment by AndyBPA** [comment]
@shiftkey attached - I tried pushing > fail > run command to clear it > pushed ok > switch branch back and forth > fails again.

[2017-09-21.desktop.production.txt](https://github.com/desktop/desktop/files/1319825/2017-09-21.desktop.production.txt)

**Comment by shiftkey** [comment]
@AndyBPA excellent, thanks for that.

```
E:\Documents\GitHub\brutes>dir .git\hooks
 Volume in drive E is Storage
 Volume Serial Number is DC6B-2390

 Directory of E:\Documents\GitHub\brutes\.git\hooks

21/09/2017  02:43 p.m.    <DIR>          .
21/09/2017  02:43 p.m.    <DIR>          ..
10/08/2017  07:14 a.m.               478 applypatch-msg.sample
10/08/2017  07:14 a.m.               896 commit-msg.sample
21/09/2017  02:43 p.m.               280 post-checkout
21/09/2017  02:43 p.m.               276 post-commit
21/09/2017  02:43 p.m.               274 post-merge
10/08/2017  07:14 a.m.               189 post-update.sample
10/08/2017  07:14 a.m.               424 pre-applypatch.sample
10/08/2017  07:14 a.m.             1,642 pre-commit.sample
21/09/2017  02:43 p.m.               270 pre-push
10/08/2017  07:14 a.m.             1,348 pre-push.sample
10/08/2017  07:14 a.m.             4,951 pre-rebase.sample
10/08/2017  07:14 a.m.               544 pre-receive.sample
10/08/2017  07:14 a.m.             1,239 prepare-commit-msg.sample
10/08/2017  07:14 a.m.             3,610 update.sample
              14 File(s)         16,421 bytes
               2 Dir(s)  800,625,565,696 bytes free
```

It looks like you have other hooks files floating around. `post-checkout` is what runs when you switch branches, and that runs `git lfs post-checkout`, which is probably restoring the hooks that it needs.

So, to _really_ clean this up, let's just remove all the hooks in the repository using `cmd`:

```
del .git\hooks\*
```

**Comment by shiftkey** [comment]
@t7m could you please open a new issue with repro steps - maybe there's something specific to that repository that we haven't encountered yet around LFS support?

**Comment by AndyBPA** [comment]
@shiftkey Hiya, tried 'del .git\hooks\*' - doesn't seem to fix it entirely, It works initially, but still get the same issue after switching to another branch and back again. 

 Directory of E:\Documents\GitHub\brutes\.git\hooks

21/09/2017  04:45 p.m.    <DIR>          .
21/09/2017  04:45 p.m.    <DIR>          ..
21/09/2017  04:45 p.m.               280 post-checkout
21/09/2017  04:45 p.m.               276 post-commit
21/09/2017  04:45 p.m.               274 post-merge
21/09/2017  04:45 p.m.               270 pre-push
               4 File(s)          1,100 bytes
               2 Dir(s)  801,440,903,168 bytes free

So it definitely feels like some sort of caching issue or something, with switching branches recovering the 'bad settings'.

If its failing to get the credentials, does it mean that they have become corrupted somehow? Are they stored anywhere that we can clear? i.e. delete my git account locally, and re-login? or not related at all? maybe a file permission issue?

[2017-09-21.desktop.production.txt](https://github.com/desktop/desktop/files/1319988/2017-09-21.desktop.production.txt)

**Comment by shiftkey** [comment]
@AndyBPA actually, before I go further, could you confirm that your repository is LFS enabled? It might have a good reason to put those hooks in, and we need to handle this error better:

```
2017-09-21T01:19:40.407Z - error: [ui] `git -c credential.helper= push origin master:master --progress` exited with an unexpected code: 1.
Remote "origin" does not support the LFS locking API. Consider disabling it with:
  $ git config lfs.https://github.com/AndyBPA/brutes.git/info/lfs.locksverify false
```

Could you try running that command to see if that makes the error go away?

**Comment by AndyBPA** [comment]
Yes we're using LFS.

Run which command? 

del .git\hooks* solves it initially, until I switch branches and back again, then I get the credential fail again.

**Comment by shiftkey** [comment]
Apologies for not being clear. This command is the one I'm interested in:

```
git config lfs.https://github.com/AndyBPA/brutes.git/info/lfs.locksverify false
```

If that works to unblock you, I can then have a chat with the LFS team about how to make this experience better.

**Comment by AndyBPA** [comment]
Ok, so I tried to push > blocked > ran your command > unblocked and pushed ok

Switched branches and back again > tried to push and it worked.

So that command seemed to fix it.

**Comment by shiftkey** [comment]
@AndyBPA thanks for confirming the workaround. I've opened #2812 to track properly addressing this.

**Comment by shiftkey** [comment]
As this issue has spiralled a bit, I'm going to lock this from spiralling out of control. Please open a new issue with details about your problem if you encounter something similar.
